llvm/lib/Target/X86/X86ISelLoweringCall.cpp

```diff
@@ -2098,15 +2153,18 @@ X86TargetLowering::LowerCall(TargetLowering::CallLoweringInfo &CLI,
       isTailCall = false;
   }
 
-  if (isTailCall && !IsMustTail) {
+  if (isTailCall) {
     // Check if it's really possible to do a tail call.
-    isTailCall = IsEligibleForTailCallOptimization(CLI, CCInfo, ArgLocs,
-                                                   IsCalleePopSRet);
+    IsSibcall = IsEligibleForTailCallOptimization(CLI, CCInfo, ArgLocs,
+                                                  IsCalleePopSRet);
```

Comment:
these changes to the sibcall logic are still needed to optimize to just a `jmp` in the case where arguments are just passed on (i.e. no stack shuffling is needed).